internal/installation/storage/storage.go

```diff
@@ -108,12 +108,24 @@ func CachePath() string {
 		cachePath = path
 	} else {
 		cachePath = configdir.New(constants.InternalConfigNamespace, "").QueryCacheFolder().Path
-		if runtime.GOOS == "windows" {
-			// Explicitly append "cache" dir as the cachedir on Windows is the same as the local appdata dir (conflicts with config)
-			cachePath = filepath.Join(cachePath, "cache")
+	}
+
+	configRoot := filepath.Dir(cachePath)
+	entries, err := os.ReadDir(configRoot)
+	if err == nil {
+		for _, e := range entries {
+			if strings.EqualFold(e.Name(), constants.InternalConfigNamespace) {
+				cachePath = filepath.Join(configRoot, e.Name())
+				break
+			}
```

Comment:
This is all to receive the case sensitive variant? It feels a bit expensive and error-prone having to list the parent dir and then retroactively get the right entry based on a string match.

Reply:
See [this](https://github.com/ActiveState/cli/pull/1752/commits/2df01e5ef43280ec0ef5885884d5a432489c14aa) commit for a rough alternate approach using `filepath.Glob`.

My guess is that attempting to determine the case of a directory on a case-insensitive filesystem will likely amount to searching.

We could dumb this down and change `InternalConfigNamespace` to `"ActiveState"` or I could track down what variable is being used to create the config directory in the first place though that would only take for new installs of the state tool.

If you have any ideas for alternatives let me know.

Reply:
Have you tried to see if using things like `filepath.Abs()` and `filepath.Clean()` might give us the case sensitive variant? And also the `path.*` variants.

Reply:
Yes, unfortunately, no easy wins there. I can certainly spend some more time digging to see if there is a better way. This approach was just one of the easier ones that came to mind.

Reply:
Ok. Can we make a helper function (eg. `fileutils.CaseSensivePath()`) and only utilize it for the specific INSTALLLDIR use-case?